snakesist/exist_client.py

```diff
@@ -278,7 +351,7 @@ def retrieve_resource(
             result_node = self.query(
                 query_expression=f"util:get-resource-by-absolute-id({abs_resource_id})"
             )
-        return result_node.xpath("./*")[0].detach()
+        return result_node.xpath("./*")[0].detach()  # TODO: Performance check?
```

Comment:
i was wondering if this could be expressed more clearly at what node is targeted here, w/o wildcards.

Reply:
i did `.root.first_child.detach()` instead of the `xpath` and also changed the method to return a `Resource` as promised in the type hint.